tracks/cpp/exercises/binary-search-tree/mentoring.md

```diff
@@ -0,0 +1,20 @@
+### Reasonable solution ###
+There is an example [in the github repo](https://github.com/exercism/cpp/blob/master/exercises/binary-search-tree/example.h) for this track.
+
+### Common Suggestions ###
+ * Use std::make_unique (c++ 14)
+### Talking Points ###
+ * How to pass parameters to constructor?
+ 
+ method       | arguemnt is lvalue    | argument is rvalue | notes
```

Comment:
Format this as a Github Markdown table, like so:
```
| method       | argument is lvalue | argument is rvalue | notes |
| ------------ | ------------------ | ------------------ | ----- |
| by const ref | copy ctor          | copy ctor          |       |
...
```

It will display like this:

| method       | argument is lvalue | argument is rvalue | notes |
| ------------ | ------------------ | ------------------ | ----- |
| by const ref | copy ctor          | copy ctor          |       |